packages/mgt-components/src/components/mgt-file-list/mgt-file-list.ts

```diff
@@ -0,0 +1,606 @@
+    }
+
+    if (!this.files || this.files.length === 0) {
+      return this.renderNoData();
+    }
+
+    return this.renderTemplate('default', { files: this.files }) || this.renderFiles();
+  }
+
+  /**
+   * Render the loading state
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtFileList
+   */
+  protected renderLoading(): TemplateResult {
+    return this.renderTemplate('loading', null) || html``;
+  }
+
+  /**
+   * Render the state when no data is available
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtFileList
+   */
+  protected renderNoData(): TemplateResult {
+    return this.renderTemplate('no-data', null) || html``;
+  }
+
+  /**
+   * Render the list of files.
+   *
+   * @protected
+   * @param {*} files
+   * @returns {TemplateResult}
+   * @memberof mgtFileList
+   */
+  protected renderFiles(): TemplateResult {
+    return html`
+      <div id="file-list-wrapper" class="file-list-wrapper">
+        <ul id="file-list" class="file-list">
+          ${repeat(
+            this.files,
+            f => f.id,
+            f => html`
+              <li class="file-item">
+                ${this.renderFile(f)}
+              </li>
+            `
+          )}
+        </ul>
+        ${this.pageIterator && this.pageIterator.hasNext ? this.renderMoreFileButton() : null}
+      </div>
+    `;
+  }
+
+  /**
+   * Render an individual file.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof mgtFileList
+   */
+  protected renderFile(file: MicrosoftGraph.DriveItem): TemplateResult {
+    const view = this.itemView;
+    return (
+      this.renderTemplate('file', { file }) ||
+      html`
+        <mgt-file .fileDetails=${file} .view=${view}></mgt-file>
+      `
+    );
+  }
+
+  /**
+   * Render the button when clicked will show more files.
+   *
+   * @protected
+   * @returns {TemplateResult}
+   * @memberof MgtFileList
+   */
+  protected renderMoreFileButton(): TemplateResult {
+    if (this.isLoadingState) {
+      return html`
+        <mgt-spinner></mgt-spinner>
+      `;
+    } else {
+      return html`<a id="show-more" class="show-more" @click=${() =>
+        this.renderNextPage()}><span>Show more items<span></a>`;
+    }
+  }
+
+  /**
+   * load state into the component.
+   *
+   * @protected
+   * @returns
+   * @memberof MgtFileList
+   */
+  protected async loadState() {
+    const provider = Providers.globalProvider;
+    if (!provider || provider.state === ProviderState.Loading) {
+      return;
+    }
+
+    if (provider.state === ProviderState.SignedOut) {
+      this.files = null;
+      return;
+    }
+    const graph = provider.graph.forComponent(this);
+    let files: MicrosoftGraph.DriveItem[];
+    let pageIterator: GraphPageIterator<MicrosoftGraph.DriveItem>;
+
+    const getFromMyDrive = !this.driveId && !this.siteId && !this.groupId && !this.userId;
+
+    if (
+      (this.driveId && (!this.itemId && !this.itemPath)) ||
```

Comment:
I see this a few times: `(!this.itemId && !this.itemPath)`. Might be nice to de-duplicate:
```
if (!this.itemId && !this.itemPath && (this.driveId || this.groupId || this.siteId || (userId && !this.insightType)) {
```

Reply:
I agree with you for code simplicity. I do think the current implementation if more readable in understanding why those are listed as is. These are the requirements of the Api endpoint of what combination of attributes are needed.

Reply:
Gotcha, I didn't realize there was a strong correlation. Might be worth commenting that :)

Reply:
Yes! Adding comments now.